src/plugins/camera/camera_definition.cpp

```diff
@@ -529,20 +529,6 @@ bool CameraDefinition::set_setting(
         return false;
     }
 
-    // FIXME: this is a hack because some params have the wrong type
-    //        when they come from the camera.
-    MAVLinkParameters::ParamValue changed_value = value;
-    if (name.compare("CAM_MODE") == 0 || name.compare("CAM_COLORMODE") == 0 ||
-        name.compare("CAM_EXPMODE") == 0 || name.compare("CAM_METERING") == 0 ||
-        name.compare("CAM_PHOTOFMT") == 0 || name.compare("CAM_PHOTOQUAL") == 0 ||
-        name.compare("CAM_VIDRES") == 0 || name.compare("CAM_WBMODE") == 0 ||
-        name.compare("CAM_COLORENCODE") == 0 || name.compare("CAM_FLICKER") == 0) {
-        if (changed_value.is_uint8()) {
-            uint8_t temp = changed_value.get_uint8();
-            changed_value.set_uint32(uint32_t(temp));
```

Comment:
But removing this will cause segfaults with H520, right?

Reply:
It might not segfault anymore because of these checks:
https://github.com/mavlink/MAVSDK/blob/1f784fec29c0a04877c6fa9b7d1213d3ac6a0a1f/src/core/mavlink_parameters.h#L416-L419

Reply:
I haven't seen such warning on my H520 :thinking:. I'll do more testing.

Reply:
I can set `CAM_EXPMODE` and `CAM_COLORMODE` without a problem on my E90. I'm improving my script to make it easier to test others, but I haven't seen segfaults until now.

Reply:
Sounds good!

Reply:
Tested with `CAM_PHOTOQUAL`, `CAM_PHOTOFMT`, I think it works.